frontend/ui/widget/virtualkeyboard.lua

```diff
@@ -310,6 +310,19 @@ function VirtualKey:init()
     self.flash_keyboard = G_reader_settings:nilOrTrue("flash_keyboard")
 end
 
+function VirtualKey:paintTo(...)
+    InputContainer.paintTo(self, ...)
+
+    -- Fudge self.dimen to include the padding, to make sure said padding is covered by our ges_events range...
+    -- Like Geom, floor coordinates & ceil dims, to fill the gaps without overlaps.
+    local coords_padding = math.floor(self.keyboard.key_padding / 2)
+    local dims_padding = coords_padding + math.ceil(self.keyboard.key_padding / 2)
```

Comment:
Isn't that similar to simpler: `local dims_padding = self.keyboard.key_padding` ?

Reply:
Yup. Artifact from the first attempt , `math.ceil(self.keyboard.key_padding / 2) * 2`,n which wasn't quite right ;p.

Reply:
(Not certain if key_padding can change, but if not, you could also do the maths only once or hardcode that in VirtualKey = ...extext{)... but probably not needed, there's only a few dozens of them.)

Reply:
Yeah, thought about it, then came to the same conclusion ;).